tutorials/misc/state_design_pattern.rst

```diff
@@ -0,0 +1,254 @@
+.. _doc_state_design_pattern:
+
+State Design Pattern
+====================
+
+Introduction
+------------
+
+Scripting a game can be difficult when there are many states that need to handled, but
+only one script can be attached to a node at a time. Instead of creating a state machine
+within the player's control script, it would make development simpler if the states were
+separated out into different classes.
+
+There are many ways to implement a state machine with godot, and some other methods are below:
+
+* The player can have a child node for each state, which are called when utilized
+* Enums can be used in conjunction with a match statement
+* The state scripts themselves could be swapped out from a node dynamically at run-time 
+
+This tutorial will focus only on adding and removing nodes which have a state script attached. Each state
+script will be an implementation of a different state.
+
+.. note::
+  There is a great resource explaining the concept of the state design pattern here:
+  https://gameprogrammingpatterns.com/state.html
+
+Script-Setup
+------------
+
+The feature of inheritance is useful for getting started with this design principle.
+A class should be created that describes the base features of the player. For now, a
+player will be limited to two actions: **move left**, **move right**. This means
+there will be two states: **idle** and **run**.
+
+Below is the generic state, from which all other states will inherit.
+
+.. tabs::
+  .. code-tab:: gdscript GDScript
+
+    extends Node2D
+    
+    class_name State
+
+    var change_state
+    var animated_sprite
+    var persistent_state
+    var velocity
+
+    func _physics_process(delta):
```

Comment:
Use `_delta`, since you're not using `delta` it would generate a warning about unused variable, which a `_` prefix prevents.